app/scripts/models/attached-clients.js

```diff
@@ -3,21 +3,52 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 /**
- * A collection of devices
+ * Attached clients are OAuth apps and devices
+ *
+ * It sorts items in order that is defined in FxA-89 feature description.
  */
-
-
 define(function (require, exports, module) {
   'use strict';
 
   var Backbone = require('backbone');
+  var Constants = require('lib/constants');
   var Device = require('models/device');
+  var OAuthApp = require('models/oauth-app');
+  var P = require('lib/promise');
+
+  var AttachedClients = Backbone.Collection.extend({
+    initialize: function (models, options = {}) {
+    },
+
+    model: function(attrs, options) {
+      if (attrs.clientType === Constants.CLIENT_TYPE_DEVICE) {
+        return new Device(attrs, options);
+      } else if (attrs.clientType === Constants.CLIENT_TYPE_OAUTH_APP) {
+        return new OAuthApp(attrs, options);
+      }
+    },
+
+    removeClient: function (itemId, user) {
+      var item = this.get(itemId);
+      return user.destroyAccountClient(user.getSignedInAccount(), item)
+        .then(() => {
+          return item;
+        });
+    },
 
-  var Devices = Backbone.Collection.extend({
-    model: Device,
+    fetchClients: function (clientTypes = {}, user) {
+      var account = user.getSignedInAccount();
+      var fetchItems = [];
 
-    initialize: function (models, options) {
-      options = options || {};
+      if (clientTypes.devices) {
```

Comment:
I like having both of the fetches in this model, it makes the view much more clean.